File: src/main/scala/werger/ports/MachineTranslationSource.scala
package werger.ports

import cats.effect.IO
import werger.domain.Language

sealed trait MtError
object MtError:
  final case class Transient(message: String) extends MtError
  final case class Unsupported(message: String) extends MtError

/**
 * Adapts one hosted machine-translation provider. Never creates a `Submission` or moves a `WorkItemStatus` — callers
 * decide what to do with a translated batch.
 */
trait MachineTranslationSource:
  def translateBatch(language: Language, sourceTexts: List[String]): IO[Either[MtError, Map[String, String]]]

File: src/main/java/werger/ports/MachineTranslationSource.java
package werger.ports;

import java.util.List;
import java.util.Map;
import werger.domain.Language;

/**
 * Adapts one hosted machine-translation provider. Never creates a {@code Submission} or moves a {@code WorkItemStatus}
 * — callers decide what to do with a translated batch.
 */
public interface MachineTranslationSource {
    Result<MtError, Map<String, String>> translateBatch(Language language, List<String> sourceTexts);
}
